language-plutus-core/stdlib/Language/PlutusCore/StdLib/Data/Function.hs

```diff
@@ -303,6 +333,16 @@ fixN n = runQuote $ do
               ]
     pure (fixNTerm, fixNType)
 
+-- | Get the fixed-point of a single recursive function.
+getSingleFixOf
+    :: (TermLike term TyName Name)
+    => ann -> term ann -> FunctionDef term TyName Name ann -> term ann
+getSingleFixOf ann fix1 fun =
+    let (FunctionDef _ _ (FunctionType _ dom cod) _) = fun
+        instantiatedFix = mkIterInst ann fix1 [dom, cod]
+        abstractedBody = mkIterLamAbs [functionDefVarDecl fun] $ _functionDefTerm fun
```

Comment:
Nit: why not just match on `fun` in the clause line?

Reply:
Because I refactored it to this point. Will do.